### PR TITLE
Homework

### PR DESCRIPTION
**Pull request by WojtekMs**
- InvalidGear exception added
- HorsePower type added to interface
- cubic centimetre unit added to interface
- Gear unit  added to interface
- ampere hour unit added to interface
- KMH unit added to interface
- accelerate / brake converted into changeSpeedTo(KMH)
- Litre unit added to interface
- encapsulation fixed & getters added
- tests added

### CODE REVIEW

File: ElectricEngine.hpp [hunk 1]
```diff
@@ -1,11 +1,20 @@
...
-class ElectricEngine
-{
-public:
-    ElectricEngine(int power, int batteryCapacity);
+#include "HorsePower.hpp"
 
-    int power_;             // in HP 
-    int batteryCapacity_;   // in Ah
+class AmpereHour {
+   public:
```
Comment (MaciejSalamonski) [inline ElectricEngine.hpp:6]: +1

**Review by MaciejSalamonski — APPROVED** [review]
Very nice! Have a good day!

File: PetrolEngine.cpp [hunk 2]
```diff
@@ -1,20 +1,22 @@
...
 }
 
-void PetrolEngine::changeGear(int gear)
-{
-    // TODO: Add checking if gear is between -1 and gears_
-    // -1 is for REAR
-    // 0 is for NEUTRAL
+void PetrolEngine::changeGear(Gear gear) {
+    if (gear.value_ < -1 || gear.value_ > gears_.value_) {
+        return;
```
Comment (WBigaj) [inline PetrolEngine.cpp:13]: Why "return"? Shouldn't here be an exception?

File: Vehicle.cpp [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        case Direction::RIGHT:
+            if (currentDirection_ == Direction::LEFT) {
+                return Direction::UP;
+            } else if (currentDirection_ == Direction::UP) {
+                return Direction::RIGHT;
+            } else if (currentDirection_ == Direction::RIGHT) {
+                return Direction::DOWN;
+            } else if (currentDirection_ == Direction::DOWN) {
+                return Direction::LEFT;
+            }
```
Comment (WBigaj) [inline Vehicle.cpp:25]: You can shorten "else if"s to "if"s ,since return escapes function anyway.

**Review by WBigaj — COMMENTED** [review]
I have checked your code and left my feedback.

**Review by Kasperice — APPROVED** [review]
Nice code, I like units you used and driving implementation ;)

**Review by ziobron — APPROVED** [review]
Nice code. Domain types are cool :)

**Comment by ziobron** [comment]
4 XP za wyjątek InvalidGear w Cars (3521)
8 * 2 XP za poprawkę interfejsu w Cars (3522)
10 XP za testy w Cars (3523)
3 * 2 XP za punktualność (1119)
{ "requirement_ids": [3521, 3523, 3522, 3522, 3522, 3522, 3522, 3522, 3522, 3522, 1119, 1119, 1119] }

**Pull request closed without merging**